cluster/src/main/java/io/scalecube/cluster/ClusterImpl.java

```diff
@@ -351,4 +352,66 @@ public NetworkEmulator networkEmulator() {
   public boolean isShutdown() {
     return onShutdown.isDisposed();
   }
+
+  public interface ClusterMonitorMBean {
+    String getMember();
+    int getIncarnation();
+    Map<String, String> getMetadata();
+    List<String> getAliveMembers(); // all alive members except local one
+    List<String> getDeadMembers(); // let's keep recent N dead members for full-filling this method
+    List<String> getSuspectedMembers();
+  }
+
+  public static class ClusterMonitor implements ClusterMonitorMBean {
+
+    private final ClusterImpl cluster;
+    private final JmxMembershipProvider membershipProvider;
+
+    public ClusterMonitor(ClusterImpl cluster) {
+
+      //this.cluster = ClusterImpl.this;
+      this.cluster = cluster;
+
+      MBeanServer server = ManagementFactory.getPlatformMBeanServer();
+      try {
+
+        membershipProvider = new JmxMembershipProvider(cluster,
+                                                       cluster.membership);
+        server.registerMBean(this, new ObjectName("io.scalecube.cluster:name=Cluster{"+cluster.member().id()+"}"));
+      } catch (Exception ex){
+        throw new RuntimeException(ex);
```

Comment:
`Exceptions.propagate(ex)` instead of `new RuntimeException(ex)`

Reply:
done